funnel/utils/markdown/base.py

```diff
@@ -32,63 +33,88 @@
 
 
 @overload
-def markdown(
-    text: None,
-    extensions: Union[List[str], None] = None,
-    extension_configs: Optional[Dict[str, str]] = None,
-    # TODO: Extend to accept helpers.EXT_CONFIG_TYPE (Dict)
-) -> None:
+def markdown(text: None, profile: Union[str, Dict[str, Any]]) -> None:
     ...
 
 
 @overload
-def markdown(
-    text: str,
-    extensions: Union[List[str], None] = None,
-    extension_configs: Optional[Dict[str, str]] = None,
-    # TODO: Extend to accept helpers.EXT_CONFIG_TYPE (Dict)
-) -> Markup:
+def markdown(text: str, profile: Union[str, Dict[str, Any]]) -> Markup:
     ...
 
 
 def markdown(
-    text: Optional[str],
-    extensions: Union[List[str], None] = None,
-    extension_configs: Optional[Dict[str, str]] = None,
-    # TODO: Extend to accept helpers.EXT_CONFIG_TYPE (Dict)
+    text: Optional[str], profile: Union[str, Dict[str, Any]]
 ) -> Optional[Markup]:
     """
     Markdown parser compliant with Commonmark+GFM using markdown-it-py.
 
-    :param bool linkify: Whether to convert naked URLs into links
-    :param list extensions: List of Markdown extensions to be enabled
-    :param dict extension_configs: Config for Markdown extensions
+    :param str|dict profile: Config profile to use
     """
     if text is None:
         return None
 
     # Replace invisible characters with spaces
     text = normalize_spaces_multiline(text)
 
-    md = MarkdownIt(
-        'gfm-like',
-        {
-            'breaks': True,
-            'linkify': True,
-            'typographer': True,
-        },
-    ).enable(['smartquotes'])
-
-    md.linkify.set({'fuzzy_link': False, 'fuzzy_email': False})
-
-    if extensions is None:
-        extensions = default_markdown_extensions
-
-    for e in extensions:
-        if e in markdown_extensions:
-            ext_config = markdown_extensions[e].default_config
-            if extension_configs is not None and e in extension_configs:
-                ext_config = markdown_extensions[e].config(extension_configs[e])
-            md.use(markdown_extensions[e].ext, **ext_config)
-
-    return Markup(md.render(text))  # type: ignore[arg-type]
+    if isinstance(profile, str):
+        try:
+            _profile = profiles[profile]
+        except KeyError as exc:
+            raise KeyError(
+                f'Wrong markdown config profile "{profile}". Check name.'
+            ) from exc
+    elif isinstance(profile, dict):
+        _profile = profile
+    else:
+        raise TypeError('Wrong type - profile has to be either str or dict')
+
+    args = _profile.get('args', ())
+
+    md = MarkdownIt(*args)
+
+    funnel_config = _profile.get('funnel_config', {})
+
+    if md.linkify is not None:
+        md.linkify.set({'fuzzy_link': False, 'fuzzy_email': False})
+
+    for action in ['enableOnly', 'enable', 'disable']:
+        if action in funnel_config:
+            getattr(md, action)(funnel_config[action])
```

Comment:
This effectively means `funnel_config` is not a TypedDict as there is no strong link between what it contains and what happens in this function. This could be better.